src/earthdataVarinfo/handler.py

```diff
@@ -0,0 +1,79 @@
+import os
+import traceback
+import json
+import shutil
+from tempfile import mkdtemp
+
+from varinfo import VarInfoFromNetCDF4
+from varinfo.cmr_search import get_granules, get_granule_link, download_granule
+from varinfo.umm_var import export_all_umm_var_to_json, get_all_umm_var
+
+
+def default(event, context):
+  print('Calling earthdata-varinfo lambda')
+  # print(json.dumps(event, indent=4))
+  # Get authorization token from header
+  headers = event['headers']
+  if 'authorization' in headers:
+    token = headers['authorization'] 
+    # print('Token in python lambda:'+token)
+
+  # Get collectionConceptId from the query parameters
+  query_string_parameters = event['queryStringParameters']
+  if query_string_parameters != None and 'collection_concept_id' in query_string_parameters:
+    collection_concept_id = query_string_parameters['collection_concept_id']
+    # print('collection_concept_id:'+collectionConceptId)
+
+  # Setup CMR search url from cmrRootUrl
+  cmr_url = os.environ['cmrRootUrl']+'/search/'
+
+  temp_dir = None
+
+  try:
+    # Retrieve a list of 10 granules for the collection
+    granules = get_granules(collection_concept_id, cmr_env=cmr_url, token=token)
+
+    # Get the URL for the first granule (NetCDF-4 file):
+    granule_link = get_granule_link(granules)
+    print('Downloading granule from '+granule_link)
+
+    # Make a temporary directory (this may cause permission issues?):
+    temp_dir = mkdtemp()
+
+    # Download file to lambda runtime environment
+    local_granule = download_granule(granule_link, token, out_directory=temp_dir)
+    print('Downloaded!')
+
+    # Parse the granule with VarInfo:
+    var_info = VarInfoFromNetCDF4(local_granule)
+
+    # Generate all the UMM-Var records:
+    all_variables = get_all_umm_var(var_info)
+
+    # Delete temporary directory and files
+    shutil.rmtree(temp_dir)
+  except:
+    # Delete temporary directory and files if created
+    if temp_dir is not None:
+      shutil.rmtree(temp_dir)
+
+    # Print error message and stacktrace
+    print('Error executing earthdata-varinfo lambda')
+    tb = traceback.format_exc()
+    print(tb)
+    return {
+      'statusCode': 400,
```

Comment:
This should be examined, GraphQL expects 200 response codes, so I'm curious how returning a non-200 will function within the resolver, it may just ignore it but we need to ensure this error message is handled by Apollo Server correctly.

Reply:
Introducing an error, I was able to get the following back in Apollo:
{
  "errors": [
    {
      "message": "400: Bad Request",
      "locations": [
        {
          "line": 4,
          "column": 5
        }
      ],
      "path": [
        "collection",
        "generateVariableDrafts"
      ],
      "extensions": {
        "response": {
          "url": "http://localhost:3013/dev/earthdataVarinfo/?collection_concept_id=C1598621093-GES_DISC",
          "status": 400,
          "statusText": "Bad Request",
          "body": {
            "Error": "Error executing earthdata-varinfo lambda",
            "StackTrace": "Traceback (most recent call last):\n  File \"/Users/wmvalenc/test/github/cmr-graphql/.venv/lib/python3.10/site-packages/urllib3/connection.py\", line 203, in _new_conn\n    sock = connection.create_connection(\n  File \"/Users/wmvalenc/test/github/cmr-graphql/.venv/lib/python3.10/site-packages/urllib3/util/connection.py\", line 60, in create_connection\n    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):\n  File ...
          }
        },
        "code": "INTERNAL_SERVER_ERROR",
        "stacktrace": [
          "GraphQLError: 400: Bad Request",
          "    at CollectionVariableDrafts.errorFromResponse (/Users/wmvalenc/test/github/cmr-graphql/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:66:21)",
          "    at CollectionVariableDrafts.didReceiveResponse (/Users/wmvalenc/test/github/cmr-graphql/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:36:30)",
          "    at /Users/wmvalenc/test/github/cmr-graphql/node_modules/@apollo/datasource-rest/dist/RESTDataSource.js:137:39",
          "    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
        ]
      }
    }
  ],
  "data": {
    "collection": {
      "conceptId": "C1598621093-GES_DISC",
      "generateVariableDrafts": null
    }
  }
}